File: interpreter/src/pregen/scala/org/enso/interpreter/nodes/expression/ParenthesesNode.scala
package org.enso.interpreter.nodes.expression

import com.oracle.truffle.api.frame.VirtualFrame
import com.oracle.truffle.api.nodes.Node.Child
import com.oracle.truffle.api.nodes.NodeInfo
import org.enso.interpreter.nodes.ExpressionNode

@NodeInfo(
  shortName   = "Parentheses",
  description = "A representation of parenthesised expressions."
)
class ParenthesesNode(@Child expression: ExpressionNode)
    extends ExpressionNode {

  override def execute(frame: VirtualFrame): Any =
    expression.execute(frame)
}

File: interpreter/src/main/java/org/enso/interpreter/node/expression/ParenthesesNode.java
package org.enso.interpreter.node.expression;

import com.oracle.truffle.api.frame.VirtualFrame;
import com.oracle.truffle.api.nodes.NodeInfo;
import org.enso.interpreter.node.ExpressionNode;

@NodeInfo(shortName = "Parentheses", description = "A representation of parenthesised expressions.")
public class ParenthesesNode extends ExpressionNode {
  @Child ExpressionNode expression;

  @Override
  public Object executeGeneric(VirtualFrame frame) {
    return expression.executeGeneric(frame);
  }
}
